### PR TITLE
Allow work with kubernetes clusters (issue #66)

### LINKED ISSUE #66: Allow work with cluster kubernetes on user-mode
# Description
New _Cloud Providers_ commands:
imagen
- **cio cloud cloud-providers list-cluster-plans** > This action lists the cluster plans offered by the cloud provider identified by the given id
	GET	/v3/cloud/cloud_providers/:cloud_provider_id/cluster_plans

New _Cloud Realms_ commands:
imagen

**cio cloud realms \<CMD>**:
imagen
- **list** > Lists all realms of a cloud provider
	GET /v3/cloud/cloud_providers/:cloud_provider_id/realms

- **show** > Shows information about the realm identified by the given id
	GET	/v3/cloud/realms/:realm_id
	
- **list-node-pool-plans** > This action lists the node pool plans offered by the realm identified by the given id
	GET	/v3/cloud/realms/:realm_id/node_pool_plans

New _Brownfield_ import command:
imagen
- **cio brownfield import-k8s-clusters**: Import kubernetes clusters for a given cloud account id
	PUT	/v3/brownfield/cloud_accounts/:cloud_account_id/import_kubernetes_clusters

New _Kubernetes_ commands:
imagen
imagen

**cio kubernetes clusters \<CMD>**:
imagen
- **list** > Lists all existing clusters
	GET /v3/kubernetes/clusters

- **show** > Shows information about the cluster identified by the given id
	GET	/v3/kubernetes/clusters/:cluster_id

- **create** > Creates a new cluster
	POST /v3/kubernetes/clusters

- **update** > Updates an existing cluster identified by the given id
	PUT /v3/kubernetes/clusters/:cluster_id

- **delete** > Deletes a cluster
	DELETE /v3/kubernetes/clusters/:cluster_id

- **retry** > Retries the application of cluster identified by the given id
	PUT /v3/kubernetes/clusters/:cluster_id/retry

- **discard** > Discards a cluster but does not delete it from the cloud provider
	DELETE /v3/kubernetes/clusters/:cluster_id/discard

- **show-plan** > Shows information about a specific cluster plan identified by the given id
	GET /v3/kubernetes/cluster_plans/:cluster_plan_id

	_Cluster is labelable_
- **add-label** > This action assigns a single label from a single labelable resource

- **remove-label** > This action unassigns a single label from a single labelable resource

**cio kubernetes node-pools \<CMD>**:
imagen
- **list** > Lists all existing node pools in a cluster
	GET /v3/kubernetes/clusters/:cluster_id/node_pools

- **show** > Shows information about the node pool identified by the given id
	GET /v3/kubernetes/node_pools/:node_pool_id

- **create** > Creates a new node pool
	POST /v3/kubernetes/clusters/:cluster_id/node_pools

- **update** > Updates an existing node pool identified by the given id
	PUT /v3/kubernetes/node_pools/:node_pool_id

- **delete** > Deletes a node pool
	DELETE /v3/kubernetes/node_pools/:node_pool_id

- **retry** > Retries the application of node pool identified by the given id
	PUT /v3/kubernetes/node_pools/:node_pool_id/retry

- **show-plan** > Shows information about a specific node pool plan identified by the given id
	GET /v3/kubernetes/node_pool_plans/:node_pool_plan_id

# Expected Behavior

# Current Behavior

# Possible Solution

# Steps to Reproduce (for bugs)

    1.
    2.
    3.
    4.

# Context and environment

### PR DESCRIPTION
**Pull request by pcantera**
**Update Go to 1.16**

# Description
New _Cloud Providers_ commands:
imagen
- **cio cloud cloud-providers list-cluster-plans** > This action lists the cluster plans offered by the cloud provider identified by the given id
	GET	/v3/cloud/cloud_providers/:cloud_provider_id/cluster_plans

New _Cloud Realms_ commands:
imagen

**cio cloud realms \<CMD>**:
imagen
- **list** > Lists all realms of a cloud provider
	GET /v3/cloud/cloud_providers/:cloud_provider_id/realms

- **show** > Shows information about the realm identified by the given id
	GET	/v3/cloud/realms/:realm_id
	
- **list-node-pool-plans** > This action lists the node pool plans offered by the realm identified by the given id
	GET	/v3/cloud/realms/:realm_id/node_pool_plans

New _Brownfield_ import command:
imagen
- **cio brownfield import-k8s-clusters**: Import kubernetes clusters for a given cloud account id
	PUT	/v3/brownfield/cloud_accounts/:cloud_account_id/import_kubernetes_clusters

New _Kubernetes_ commands:
imagen
imagen

**cio kubernetes clusters \<CMD>**:
imagen
- **list** > Lists all existing clusters
	GET /v3/kubernetes/clusters

- **show** > Shows information about the cluster identified by the given id
	GET	/v3/kubernetes/clusters/:cluster_id

- **create** > Creates a new cluster
	POST /v3/kubernetes/clusters

- **update** > Updates an existing cluster identified by the given id
	PUT /v3/kubernetes/clusters/:cluster_id

- **delete** > Deletes a cluster
	DELETE /v3/kubernetes/clusters/:cluster_id

- **retry** > Retries the application of cluster identified by the given id
	PUT /v3/kubernetes/clusters/:cluster_id/retry

- **discard** > Discards a cluster but does not delete it from the cloud provider
	DELETE /v3/kubernetes/clusters/:cluster_id/discard

- **show-plan** > Shows information about a specific cluster plan identified by the given id
	GET /v3/kubernetes/cluster_plans/:cluster_plan_id

	_Cluster is labelable_
- **add-label** > This action assigns a single label from a single labelable resource

- **remove-label** > This action unassigns a single label from a single labelable resource

**cio kubernetes node-pools \<CMD>**:
imagen
- **list** > Lists all existing node pools in a cluster
	GET /v3/kubernetes/clusters/:cluster_id/node_pools

- **show** > Shows information about the node pool identified by the given id
	GET /v3/kubernetes/node_pools/:node_pool_id

- **create** > Creates a new node pool
	POST /v3/kubernetes/clusters/:cluster_id/node_pools

- **update** > Updates an existing node pool identified by the given id
	PUT /v3/kubernetes/node_pools/:node_pool_id

- **delete** > Deletes a node pool
	DELETE /v3/kubernetes/node_pools/:node_pool_id

- **retry** > Retries the application of node pool identified by the given id
	PUT /v3/kubernetes/node_pools/:node_pool_id/retry

- **show-plan** > Shows information about a specific node pool plan identified by the given id
	GET /v3/kubernetes/node_pool_plans/:node_pool_plan_id

# Related Issue
Closes #66 

# Motivation and Context

# How Has This Been Tested?

# Screenshots (if appropriate)

# Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

# Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by lttito — APPROVED** [review]
:+1:

File: api/cloud/cloud_providers_api_test.go [hunk 1]
```diff
@@ -36,3 +36,19 @@ func TestListLoadBalancerPlans(t *testing.T) {
...
+func TestListClusterPlans(t *testing.T) {
+	clusterPlansIn := testdata.GetClusterPlanData()
+	ListClusterPlansMocked(t, clusterPlansIn, clusterPlansIn[0].CloudProviderID)
+	ListClusterPlansFailErrMocked(t, clusterPlansIn, clusterPlansIn[0].CloudProviderID)
+	ListClusterPlansFailStatusMocked(t, clusterPlansIn, clusterPlansIn[0].CloudProviderID)
+	ListClusterPlansFailJSONMocked(t, clusterPlansIn, clusterPlansIn[0].CloudProviderID)
+}
+
+func TestListNodePoolPlans(t *testing.T) {
+	clusterPlansIn := testdata.GetNodePoolPlanData()
```
Comment (spatino) [inline api/cloud/cloud_providers_api_test.go:49]: This variable should be named "nodePoolPlanIn"

**Review by spatino — APPROVED** [review]
:+1:

**Review by chris-m-h — APPROVED** [review]

**Review by lttito — APPROVED** [review]
:+1:

**Review by pbanos — APPROVED** [review]

**Pull request merged**